chain-factory/src/factory.rs

```diff
@@ -30,29 +36,105 @@ pub trait FactoryModule {
         min_validators: usize,
         max_validators: usize,
         min_stake: BigUint,
+        chain_name: ManagedBuffer,
         additional_stake_required: MultiValueEncoded<StakeMultiArg<Self::Api>>,
     ) {
         let payment_amount = self.call_value().egld_value().clone_value();
         let deploy_cost = self.deploy_cost().get();
         require!(payment_amount == deploy_cost, "Invalid payment amount");
 
-        let caller = self.blockchain().get_caller();
         let source_address = self.chain_config_template().get();
+        let metadata =
+            CodeMetadata::PAYABLE_BY_SC | CodeMetadata::UPGRADEABLE | CodeMetadata::READABLE;
+        let args = self.get_deploy_chain_config_args(
+            &min_validators,
+            &max_validators,
+            &min_stake,
+            &additional_stake_required,
+        );
+
+        let sc_address = self
+            .tx()
+            .raw_deploy()
+            .gas(self.blockchain().get_gas_left())
+            .from_source(source_address)
+            .code_metadata(metadata)
+            .arguments_raw(args)
+            .returns(ReturnsNewManagedAddress)
+            .sync_call();
+
+        self.all_deployed_contracts(ManagedBuffer::new())
```

Comment:
Why do you use ManagedBuffer::new() here?

Reply:
fixed, it was supposed to be `chain_id`